auth0/provider_test.go

```diff
@@ -20,21 +22,66 @@ import (
 
 const wireMockURL = "localhost:8080"
 
-var (
-	testProviderFactories = map[string]func() (*schema.Provider, error){
+func testProviders(httpRecorder *recorder.Recorder) map[string]func() (*schema.Provider, error) {
+	return map[string]func() (*schema.Provider, error){
 		"auth0": func() (*schema.Provider, error) {
-			return Provider(), nil
+			provider := Provider()
+
+			provider.ConfigureContextFunc = configureTestProvider(httpRecorder)
+
+			return provider, nil
 		},
 	}
+}
 
-	testProviderFactoriesWithMockedAPI = map[string]func() (*schema.Provider, error){
-		"auth0": func() (*schema.Provider, error) {
-			return providerWithMockedAPI(), nil
-		},
+func configureTestProvider(
+	httpRecorder *recorder.Recorder,
+) func(ctx context.Context, data *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	return func(ctx context.Context, data *schema.ResourceData) (interface{}, diag.Diagnostics) {
+		domain := data.Get("domain").(string)
+		debug := data.Get("debug").(bool)
+
+		testClient := &http.Client{}
+		if httpRecorder != nil {
+			testClient.Transport = httpRecorder
+		}
+
+		apiClient, err := management.New(
+			domain,
+			management.WithStaticToken("insecure"),
```

Comment:
By default we want to set up our management client as insecure as we default directly to using our fake domain, and we want to skip making those oauth2 token requests, as we don't want to record those interactions.